test/test_suite.py

```diff
@@ -68,3 +83,154 @@ def test_percent_sign_round_trip(self):
+    def define_tables(cls, metadata):
+        Table(
+            "datetime_table",
+            metadata,
+            Column("id", Integer, primary_key=True, test_needs_autoincrement=True),
+            Column("date_data", cls.datatype),
+        )
+
+    def test_null(self):
+        date_table = self.tables.datetime_table
+
+        config.db.execute(date_table.insert(), {"id": 1, "date_data": None})
+
+        row = config.db.execute(select([date_table.c.date_data])).first()
+        eq_(row, (None,))
+
+    def test_round_trip(self):
+        date_table = self.tables.datetime_table
+        config.db.execute(date_table.insert(), {"id": 1, "date_data": self.data})
+
+        row = config.db.execute(select([date_table.c.date_data])).first()
+        compare = self.compare or self.data
+        compare = compare.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        eq_(row[0].rfc3339(), compare)
+        assert isinstance(row[0], DatetimeWithNanoseconds)
+
+    @requires.standalone_null_binds_whereclause
+    def test_null_bound_comparison(self):
+        # this test is based on an Oracle issue observed in #4886.
+        # passing NULL for an expression that needs to be interpreted as
+        # a certain type, does the DBAPI have the info it needs to do this.
+        date_table = self.tables.datetime_table
+        with config.db.connect() as conn:
+            result = conn.execute(
+                date_table.insert(), {"id": 1, "date_data": self.data}
+            )
+            id_ = result.inserted_primary_key[0]
+            stmt = select([date_table.c.id]).where(
+                case(
+                    [
+                        (
+                            bindparam("foo", type_=self.datatype) != None,
+                            bindparam("foo", type_=self.datatype),
+                        )
+                    ],
+                    else_=date_table.c.date_data,
+                )
+                == date_table.c.date_data
+            )
+
+            row = conn.execute(stmt, {"foo": None}).first()
+            eq_(row[0], id_)
+
+
+class DateTimeTest(_DateTimeTest, DateTimeMicrosecondsTest):
+    pass
+
+
+class DateTimeCoercedToDateTimeTest(
+    _DateTimeCoercedToDateTimeTest, DateTimeMicrosecondsTest
+):
+    def test_round_trip(self):
+        date_table = self.tables.datetime_table
+        config.db.execute(date_table.insert(), {"id": 1, "date_data": self.data})
+
+        row = config.db.execute(select([date_table.c.date_data])).first()
+        compare = self.compare or self.data
+
+        eq_(row[0].date(), compare)
+        assert isinstance(row[0], DatetimeWithNanoseconds)
+
+
+class DateTimeHistoricTest(_DateTimeHistoricTest):
+    pass
+
+
+class DateHistoricTest(_DateHistoricTest):
+    pass
+
+
+class TimestampMicrosecondsTest(_TimestampMicrosecondsTest):
+    pass
+
+
+class TimeMicrosecondsTest(_TimeMicrosecondsTest):
+    # Time data type doesn't support by python-spanner.
+    def test_null(self):
```

Comment:
My understanding is that this will cause the test to pass as it is overloading the test to be empty. This is not how we should handle unsupported tests. We should mark these tests as skipped using pytest. That way it is clear that the connector does not support these use cases and an explanation can be added with why. See #11 for an example.

Reply:
Removed empty classes and pytest skip added.